### ISSUE TITLE
[Feature] Command goi18n generate dynamic struct R.go 

### ISSUE THREAD

**Issue by rodrigocorsi2** [nicksnyder/go-i18n#50]
Working with Android/Java the IDE generates a dynamic class `R.java` with all strings ids, this method avoid a lot of bugs. In a big project is easy to reference an invalid id or to have a lot of unused ids.

What do you think the command `goi18n` generates `R.go` ?
Example:

en-US.all.json
```json
[
  {
    "id": "settings_title",
    "translation": "Settings"
  }
]
```

```bash
$ goi18n -genconst en-US.all.json -outdir ./
```
this command generates `R.go`
```go
package R
const SettingsTitle string = "settings_title"
```

In code the new way will be:
```go
T(R.SettingsTitle)
```
This is only a suggestion, let me know what do you think.
I can help with this change
thanks

**Comment by nicksnyder** [comment]
Sounds like a good idea to me! I will happily merge a PR that implements this.

Can you clarify if you think this should be a separate command or just something that happens (optionally) when goi18n is run for merging purposes?

Other thoughts

(1) Should it be possible to configure the name of "R" to be something else? R seems like a good default.
(2) Should this be opt-out instead of opt-in?

**Comment by rodrigocorsi2** [comment]
Due to compatibility with old projects, I think it should be optional using some parameters on the command goi18n.
```
$ goi18n -genconst -package=R en-US.all.json -outdir ./
```
The user can put this command in a Makefile for example.

But if you prefer more solid approach we can change `translationID` to `int`
```go
type TranslateFunc func(translationID int, args ...interface{}) string
```
In this situation the command will generate "R" like this:
```go
package R
const SettingsTitle int = 0
const OtherString  int = 1
```

**Comment by nicksnyder** [comment]
It is still unclear to me if you are proposing a separate command or not.

Would `$ goi18n -genconst -package=R en-US.all.json -outdir ./` also do the normal merge process or would it *only* generate the constants?

I don't want to change the type of translationID (it should remain a string).

**Comment by rodrigocorsi2** [comment]
I'm not proposing to create a new command but add new parameters to `goi18n`.
The command `$ goi18n -genconst -package=R en-US.all.json -outdir ./` will only generate the constants.

**Comment by nicksnyder** [comment]
Ok, the way I read it you are proposing a new command (in the same way that `git commit` is a different command than `git push`). goi18n === git, except right now it only has one implicit command (imagine if just running `git` did something useful).

There are two ways we could proceed:

1. Make proper commands for goi18n
    - `goi18n merge [same usage as before]` for the current behavior (unsure about the name `merge`).
    - `goi18n constants -package=R en-US.all.json -outdir ./` for the new behavior (only generates constants and doesn't do merge).
2. Treat constant generation as an optional part of the merge process (either opt-in or opt-out).

I think I would prefer (1).

**Comment by nicksnyder** [comment]
For the sake of backwards compatibility, if no command is explicitly provided we can default to "merge". In this case I would log a message to the console to ask users to start using "merge" explicitly.

**Comment by rodrigocorsi2** [comment]
So it seems good to me, I'll do some tests once it is ready I will submit a PR

**Comment by nicksnyder** [comment]
Great, thanks! Please do include tests as part of the PR.

**Comment by nicksnyder** [comment]
@rodrigocorsi2 merged and released in 1.5.0. Thanks!
